library/core/class.format.php

```diff
@@ -1478,9 +1478,10 @@ public static function getEmbedSize() {
      * Format a string using Markdown syntax. Also purifies the output html.
      *
      * @param mixed $Mixed An object, array, or string to be formatted.
+     * @param boolean $Flavored Optional. Parse with Vanilla-flavored settings? Default true
      * @return string
      */
-    public static function markdown($Mixed) {
+    public static function markdown($Mixed, $Flavored = true) {
```

Comment:
This implementation would require you to change every call of this function if you want to remove the flavor. I would think this should default to a configuration option instead of true.

    public static function markdown($mixed, $flavored = null) {
      if(is_null($flavored)) {
        $flavored = c('Garden.InputFormatter.Flavored', true);
      }
    ...
    }